addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SinkDetectionFindProbableSinks.java

```diff
@@ -70,18 +73,38 @@ public String getReference() {
 
     @Override
     public void scan() {
-
         HttpMessage msg = getBaseMsg();
         try {
             HttpMessage msg1 = msg.cloneRequest();
             sendAndReceive(msg1, false);
-            PersistentXssUtils.testForSink(msg1);
-
+            findProbableSinks(msg1);
         } catch (Exception e) {
             log.error(e.getMessage(), e);
         }
     }
 
+    private void findProbableSinks(HttpMessage msg) {
+        SinkDetectionStorage storage = getStorage();
+        String msgBody = msg.getResponseBody().toString();
+        Set<String> valuesSeenInResponseBody = storage.getSeenValuesContainedInString(msgBody);
+
+        for (String value : valuesSeenInResponseBody) {
```

Comment:
@DiogoMRSilva I have a small question, here we are only checking if the seen values are present in response body which is correct for XSS but for File Upload vulnerabilities causing Path traversal or XSS can have the param value present in the URL and that can be a sink for them. 
So can we add the logic where anywhere we find the seen value is a sink ? @kingthorin @thc202  @DiogoMRSilva 
 please correct me if you think otherwise.

thanks,
Karan

Reply:
I believe that searching only in the body is also the correct way for file uploads because the content of the file you uploaded will appear in the body of a response. The problem here is that the sink finder will not detect the file uploads sinks if there is some file manipulation in the backend, even if only 1 letter.

Reply:
Actually i was thinking again on this and i think if we check param in url still it will not work as `SinkDetectionVerifyProbableSinks` will fail because sink url has not changed. So we need to think some other way for file uploads.

Reply:
> I believe that searching only in the body is also the correct way for file uploads because the content of the file you uploaded will appear in the body of a response. The problem here is that the sink finder will not detect the file uploads sinks if there is some file manipulation in the backend, even if only 1 letter.

Yeah but file manipulation should not be a regular usecase. But i think you gave a very good perspective to think. thanks